addons/purchase/purchase.py

```diff
@@ -1370,9 +1383,34 @@ def make_po(self, cr, uid, ids, context=None):
                     available_po_line_ids = po_line_obj.search(cr, uid, [('order_id', '=', po_id), ('product_id', '=', line_vals['product_id']), ('product_uom', '=', line_vals['product_uom'])], context=context)
                     if available_po_line_ids:
                         po_line = po_line_obj.browse(cr, uid, available_po_line_ids[0], context=context)
-                        po_line_obj.write(cr, SUPERUSER_ID, po_line.id, {'product_qty': po_line.product_qty + line_vals['product_qty']}, context=context)
                         po_line_id = po_line.id
-                        sum_po_line_ids.append(procurement.id)
+                        uom_obj = self.pool.get('product.uom')
```

Comment:
The code takes a lot of space in the method, so we could put it in a different method.  This could be common between the make_po and the cancelling.  It could be better to use purchase_line_id.procurement_ids instead of the search query as this might still be in the cache.